com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/clients/ServiceCreator.java

```diff
@@ -54,13 +57,23 @@ public void createService(CloudFoundryOperations client, CloudServiceExtended se
     }
 
     private List<String> computeValidServiceOfferings(CloudFoundryOperations client, List<String> possibleServiceOfferings,
-        List<String> existingServiceOfferings) {
+        String servicePlan, Map<String, List<CloudServicePlan>> existingServiceOfferings) {
         List<String> validServiceOfferings = new ArrayList<String>();
         for (String possibleServiceOffering : possibleServiceOfferings) {
-            if (existingServiceOfferings.contains(possibleServiceOffering)) {
+            for (String existingServiceOffering : existingServiceOfferings.keySet()) {
+                if (!existingServiceOffering.equals(possibleServiceOffering)) {
+                    LOGGER.warn(format("Service offering \"{0}\" is not existing", possibleServiceOffering));
```

Comment:
If there are N existing offerings and the one we want is in the 4th index, wouldn't this line print the same message 4 times?

Reply:
Yes, there will be warning for each not matched service offering.

Reply:
Maybe the warning should be extended to something like "Service offering \"{0}\" does not exist in service offering \"{1}\", because having the same 4 messages in the log is not really useful IMO and is more of a distraction than help.

Reply:
Done.